<a href="https://www.kaggle.com/code/nanduvardhanreddy/data-science-jobs?scriptVersionId=115277805" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

# 1.☞IMPORTING LIBRARIES

In [1]:
import pandas as pd
import numpy as np
import plotly.express as px

In [2]:
df=pd.read_csv('../input/data-science-job-salaries/ds_salaries.csv')
df.head()

Unnamed: 0  work_year experience_level employment_type  \
0           0       2020               MI              FT   
1           1       2020               SE              FT   
2           2       2020               SE              FT   
3           3       2020               MI              FT   
4           4       2020               SE              FT   

                    job_title  salary salary_currency  salary_in_usd  \
0              Data Scientist   70000             EUR          79833   
1  Machine Learning Scientist  260000             USD         260000   
2           Big Data Engineer   85000             GBP         109024   
3        Product Data Analyst   20000             USD          20000   
4   Machine Learning Engineer  150000             USD         150000   

  employee_residence  remote_ratio company_location company_size  
0                 DE             0               DE            L  
1                 JP             0               JP            S  
2                 GB            50               GB            M  
3                 HN             0               HN            S  
4                 US            50               US            L

**WE DON'T NEED 'UNNAMED:0',  'SALARY',  'SALARY_CURRENCY' BECAUSE WE ARE GONNA USE SALARY IN USD SO LET'S REMOVE THEM**

In [3]:
df.drop(['Unnamed: 0','salary','salary_currency'],axis=1,inplace=True)
df.head()


work_year experience_level employment_type                   job_title  \
0       2020               MI              FT              Data Scientist   
1       2020               SE              FT  Machine Learning Scientist   
2       2020               SE              FT           Big Data Engineer   
3       2020               MI              FT        Product Data Analyst   
4       2020               SE              FT   Machine Learning Engineer   

   salary_in_usd employee_residence  remote_ratio company_location  \
0          79833                 DE             0               DE   
1         260000                 JP             0               JP   
2         109024                 GB            50               GB   
3          20000                 HN             0               HN   
4         150000                 US            50               US   

  company_size  
0            L  
1            S  
2            M  
3            S  
4            L

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 607 entries, 0 to 606
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           607 non-null    int64 
 1   experience_level    607 non-null    object
 2   employment_type     607 non-null    object
 3   job_title           607 non-null    object
 4   salary_in_usd       607 non-null    int64 
 5   employee_residence  607 non-null    object
 6   remote_ratio        607 non-null    int64 
 7   company_location    607 non-null    object
 8   company_size        607 non-null    object
dtypes: int64(3), object(6)
memory usage: 42.8+ KB


***NO NULL VALUES***

In [5]:
df.duplicated().value_counts()

False    565
True      42
dtype: int64

***THERE ARE 42 DUPLICATED VALUES SO LET'S REMOVE THEM***

In [6]:
df.drop_duplicates(inplace=True)

In [7]:
df.duplicated().value_counts() # SO, all duplicates are removed

False    565
dtype: int64

***LET'S CONVERT THE 2-LETTER COUNTRY NAME TO FULL NAMES FOR OUR CONVENIENCE***

In [8]:
pip install country_converter --upgrade

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.5/53.5 kB 227.0 kB/s eta 0:00:00
  Preparing metadata (setup.py) ... - done
  Created wheel for country_converter: filename=country_converter-0.8.0-py3-none-any.whl size=55658 sha256=0e1ee2cd32f1fde8e9fc13a0400b56d75a3b43c8040f4d28f53b3f3006e822bf
  Stored in directory: /root/.cache/pip/wheels/ac/ef/42/232d7f54c493023bbf8710c3ab50939d833a2e4062afd0dea3
Successfully built country_converter
Note: you may need to restart the kernel to use updated packages.


In [9]:
import country_converter as coco #THIS WILL CONVERT OUR NAMES 

***Converting company_location and employee_residence***

In [10]:
df['company_location']=coco.convert(names=df['company_location'],to='name_short')
df['employee_residence']=coco.convert(names=df['employee_residence'],to='name_short')

***REPLACING THE TEXT OUR CONVENIENCE***

In [11]:
df['experience_level']=df['experience_level'].replace(['MI','SE','EX','EN'],[' Mid-level','Senior-level','Executive-level','Entry-level'])
df['employment_type']=df['employment_type'].replace(['FT','CT','PT','FL'],['Full-time','Contract','Part-time','Freelance'])
df['company_size']=df['company_size'].replace(['S','L','M',],['Small','Large','Medium'])
df['remote_ratio']=df['remote_ratio'].replace([0,50,100],['Onsite','Hybrid','Remote'])

In [12]:
df.head()

work_year experience_level employment_type                   job_title  \
0       2020        Mid-level       Full-time              Data Scientist   
1       2020     Senior-level       Full-time  Machine Learning Scientist   
2       2020     Senior-level       Full-time           Big Data Engineer   
3       2020        Mid-level       Full-time        Product Data Analyst   
4       2020     Senior-level       Full-time   Machine Learning Engineer   

   salary_in_usd employee_residence remote_ratio company_location company_size  
0          79833            Germany       Onsite          Germany        Large  
1         260000              Japan       Onsite            Japan        Small  
2         109024     United Kingdom       Hybrid   United Kingdom       Medium  
3          20000           Honduras       Onsite         Honduras        Small  
4         150000      United States       Hybrid    United States        Large

# 2.JOBS ANALYSIS

In [13]:
experience=df['experience_level'].value_counts()
fig=px.pie(names=experience.index,
           values=experience.values,
           color=experience.index,
           title='Jobs based on Experience level',
           template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

In [14]:
remote=df['remote_ratio'].value_counts()
fig=px.pie(names=remote.index,
           values=remote.values,
           color=remote.index,
           title='Jobs based on remote',
           template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

***AS WE CAN SEE MANY COMPANIES ARE DOING WORK FROM HOME WHICH IS 61.2%***

# 3.COMAPNIES ANLYSIS

In [15]:
size=df['company_size'].value_counts()
fig=px.pie(names=size.index,
           values=size.values,
           color=size.index,
           title='COMPANIES BASED ON SIZE',
           template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

***MEDIUM SIZED COMPANIES ARE MORE THAN HALF IN THE DATA SCIENCE INDUSTRY***

In [16]:
df['same area']=df.apply(lambda x: 'same location' if x.employee_residence==x.company_location else 'different location',axis=1)
df['same area'].value_counts()

same location         514
different location     51
Name: same area, dtype: int64

***OUT OF 555 EMPLOYEES 514 ARE RESIDING IN THE SAME STATE AS THE COMPANY LOCATION***

***EMPLOYEES IN SAME AREA BUT WORKING FULLY REMOTE***

In [17]:
df['same area but remote']=df.apply(lambda x: 'yes' if (x.employee_residence==x.company_location)&(x.remote_ratio=='Remote') else 'no',axis=1)
remotly=df['same area but remote'].value_counts()
fig=px.pie(names=remotly.index,
           values=remotly.values,
           color=remotly.index,
           title='EMPLOYEES IN SAME AREA BUT WORKING FULLY REMOTE',
           template='plotly_dark')
fig.update_traces(textinfo='label+percent+value')
fig.show()

***EVEN THOUGH 308 LIVING IN THE SAME AREA AS IN THE COMPANY'S LOCATION BUT THEY ARE WORKING FULLY REMOTE I THINK COVID PLAYS MAJOR ROLE IN THIS.***

# 4.TOP 10

In [18]:
top_10_locations=df['company_location'].value_counts().head(10)

In [19]:
    top_10_employee_location=df['employee_residence'].value_counts().head(10)
    fig=px.bar(x=top_10_employee_location.index,
               y=top_10_employee_location.values,
               color=top_10_employee_location.index,
               text=top_10_employee_location.values,
               title='No.of employees per country')
    fig.update_layout(xaxis_title='country',yaxis_title='N0.of employees')

In [20]:
avg_salary_per_employee=df.groupby(by='company_location')['salary_in_usd'].mean().sort_values(ascending=False).head(10)
fig=px.bar(x= avg_salary_per_employee.index,
           y= avg_salary_per_employee.values,
           color= avg_salary_per_employee.index,
           text= avg_salary_per_employee.values,
           title=' Avg_salary_per_employee')
fig.update_layout(xaxis_title='Country',yaxis_title='Salary_in_usd')

***RUSSIANS ARE THE TOP PAID DATA PEOPLE IN THE WORLD WHICH IS 157500 USD SECOND AND THIRD ARE AMERICANS AND NEW ZEALAND ACCORDING TO OUR DATA BUT WE CANNOT TRUST THIS DATA BECAUSE ONLY 1 DATA POINT RECORDED FROM RUSSIA***

In [21]:
top_10_job_title=df['job_title'].value_counts().head(10)
fig=px.bar(x=top_10_job_title.index,
               y= top_10_job_title.values,
               color= top_10_job_title.index,
               text= top_10_job_title.values,
               title='Top job titles')
fig.update_layout(xaxis_title='Jobs',yaxis_title='N0.of Jobs')

***DATA SCIENTIST IS TOP MOST JOB_TITLES WITH 130 AND THE FOLLOWING ARE DATA ENGINEER AND DATA ANALYST WITH 121 AND 82 RESPECTIVELY***

In [22]:
Avg_salaries_per_role=df.groupby(by='job_title')['salary_in_usd'].mean().sort_values(ascending=False).head(10)
fig=px.bar(x= Avg_salaries_per_role.index,
           y=  Avg_salaries_per_role.values,
           color= Avg_salaries_per_role.index,
           text= Avg_salaries_per_role.values,
           title=' Avg_salary_per_employee')
fig.update_layout(xaxis_title='Country',yaxis_title='Salary_in_usd')

# ***DATA ANALYTICS LEAD GETS THE HIGH PAY WHICH IS 405000 USD AND NEXT IS PRINCIPAL DATA ENGINEER AND FINANCIAL DATA ENGINEER WITH 328333.33 USD AND 275000 USD***